### PR TITLE
Add SNS support in notifications plugin

### PR DESCRIPTION
**Pull request by joshuali925**
### Description
- Added SNS support to notifications. 
- This PR (and its CI) depends on common utils PR: https://github.com/opensearch-project/common-utils/pull/50
- Will send separate PR to add tests.

### Issues Resolved
- #240 
- #155 

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by zhongnansu** [comment]
Could you update your PR to pass the CI? The upstream CI failure was temporarily fixed. So PRs should pass CI now

**Comment by joshuali925** [comment]
> Could you update your PR to pass the CI? The upstream CI failure was temporarily fixed. So PRs should pass CI now

CI depends on common-utils PR, need that merged first

File: notifications/spi/build.gradle [hunk 1]
```diff
@@ -125,12 +136,15 @@ dependencies {
             'io.mockk:mockk-agent-api:1.11.0',
             'io.mockk:mockk-agent-common:1.11.0',
             'io.mockk:mockk-agent-jvm:1.11.0',
+            //'net.bytebuddy:byte-buddy:1.10.0', // required by mockk for final class mock
+            "org.jetbrains.kotlin:kotlin-reflect:${kotlin_version}", // required by mockk
```
Comment (zhongnansu) [inline notifications/spi/build.gradle:140]: this is duplicate with ine 135 below.

File: notifications/spi/build.gradle [hunk 2]
```diff
@@ -125,12 +136,15 @@ dependencies {
             'io.mockk:mockk-agent-api:1.11.0',
             'io.mockk:mockk-agent-common:1.11.0',
             'io.mockk:mockk-agent-jvm:1.11.0',
+            //'net.bytebuddy:byte-buddy:1.10.0', // required by mockk for final class mock
```
Comment (zhongnansu) [inline notifications/spi/build.gradle:139]: Remove the commented code

File: notifications/spi/build.gradle [hunk 3]
```diff
@@ -125,12 +136,15 @@ dependencies {
...
+            "org.jetbrains.kotlin:kotlin-reflect:${kotlin_version}", // required by mockk
     )
     testImplementation 'org.springframework.integration:spring-integration-mail:5.5.0'
     testImplementation 'org.springframework.integration:spring-integration-test-support:5.5.0'
     testRuntime('org.junit.jupiter:junit-jupiter-engine:5.6.2')
     testCompile "org.jetbrains.kotlin:kotlin-test:${kotlin_version}"
     testCompile "org.mockito:mockito-core:3.10.0"
+    //testCompile "org.mockito:mockito-inline:3.10.0"
```
Comment (zhongnansu) [inline notifications/spi/build.gradle:147]: remove the commented code

File: notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/client/DestinationSNSClient.kt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * This class handles the SNS connections to the given Destination.
+ */
+class DestinationSNSClient(destination: SNSDestination) {
+
+    private val amazonSNS: AmazonSNS = SNSClientFactory().getClient(destination)
+
+    fun execute(topicArn: String, message: MessageContent): String {
+        val result = amazonSNS.publish(topicArn, message.textDescription, message.title)
+        return result.messageId // TODO: return what?
```
Comment (zhongnansu) [inline notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/client/DestinationSNSClient.kt:28]: I didn't get he TODO here? Could you add more explaination

Comment (joshuali925) [reply notifications/spi/src/main/kotlin/org/opensearch/notifications/spi/client/DestinationSNSClient.kt:28]: Forgot to change this from chen's branch. Seems it's ok to return messageId here, which will be used as statusText. Will update

File: notifications/spi/build.gradle [hunk 5]
```diff
@@ -112,6 +122,7 @@ dependencies {
     implementation "com.github.seancfoley:ipaddress:5.3.3"
     testImplementation(
             'org.assertj:assertj-core:3.16.1',
+            'com.nhaarman.mockitokotlin2:mockito-kotlin:2.2.0',
```
[the remark this answers is not in the document]
Comment (joshuali925) [reply notifications/spi/build.gradle]: removed duplicates and comments

**Review by zhongnansu — APPROVED** [review]

**Pull request merged**